src/QsCompiler/LanguageServer/Utils.cs

```diff
@@ -124,4 +142,30 @@ public override void Initialize(IEventSource eventSource)
             }
         }
     }
+
+    internal static class DotNetSdkHelper
+    {
+        private static readonly Regex DotNet31Regex = new Regex(@"^3\.1\.\d+", RegexOptions.Multiline | RegexOptions.Compiled);
+
+        public static bool? IsDotNet31Installed()
+        {
+            var listSdkProcess = Process.Start(
+                new ProcessStartInfo()
+                {
+                    FileName = "dotnet",
+                    Arguments = "--list-sdks",
```

Comment:
Given that the command lists one the sdks, and I don't see that list ever getting to long this is fine here. Just as an FYI, ReadToEnd may deadlock if the output is too long.

Reply:
Doesn't that only apply if the stream never ends, not if the stream is simply too long? (It's possible a broken or malicious version of `dotnet` could output forever, though, since it's an external command.)

Reply:
No, it can be an issue even for finite streams; but for this case it should be irrelevant.